modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2RowCache.java

```diff
@@ -1,163 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.ignite.internal.processors.query.h2;
-
-import java.util.Iterator;
-import java.util.Map;
-import org.apache.ignite.IgniteCheckedException;
-import org.apache.ignite.internal.pagemem.PageIdUtils;
-import org.apache.ignite.internal.pagemem.PageMemory;
-import org.apache.ignite.internal.processors.cache.CacheGroupContext;
-import org.apache.ignite.internal.processors.cache.GridCacheContextInfo;
-import org.apache.ignite.internal.processors.query.GridQueryRowCacheCleaner;
-import org.apache.ignite.internal.processors.query.h2.opt.H2CacheRow;
-import org.apache.ignite.internal.util.typedef.F;
-import org.jsr166.ConcurrentLinkedHashMap;
-
-import static org.jsr166.ConcurrentLinkedHashMap.DFLT_INIT_CAP;
-import static org.jsr166.ConcurrentLinkedHashMap.DFLT_LOAD_FACTOR;
-
-/**
- * H2 row cache.
- */
-public class H2RowCache implements GridQueryRowCacheCleaner {
```

Comment:
Why did you decide to get rid of this? This row cache is not the same thing as H2's internal object cache

Reply:
This was discussed with @tledkov-gridgain. It was an optimization for some specific corner cases for a specific GG client. Decided to remove this code as obsolete.

Reply:
I don't sure it's a good idea to remove row cache under within this PR. If the row cache is really useless, it's better to start a discussion on the dev list first.